File: Esercitazioni/Proxy_skeleton_Java_1/src/server/DispatcherSkeleton.java
package server;

import service.IDispatcher;

import java.io.IOException;
import java.net.*;

public abstract class DispatcherSkeleton implements IDispatcher{
    private int port;

    public DispatcherSkeleton(int port){
        this.port = port;
    }

    public void runSkeleton(){
        try{
            ServerSocket serverSocket = new ServerSocket(port);
            
            while(true){
                ServerThread w = new ServerThread(serverSocket.accept(), this);
                w.start();
            }
        }catch(IOException e){
            e.printStackTrace();
        }


    }
}

File: Esercitazioni/pythonJava/python/DispatcherSkeleton.py
from IDispatcher import IDispatcher
from abc import ABC
import socket, multiprocessing


def Worker(remoteSocket:socket.socket, skeleton):
    msg = remoteSocket.recv(1024).decode()

    splitted_msg = msg.split('-')

    if('deposita' in msg):
        skeleton.deposita(int(splitted_msg[1]))
    elif('preleva' in msg):
        res = str(skeleton.preleva())
        remoteSocket.send((res+"\n").encode())
    else:
        print('[SERVER]\t richiesta non riconosciuta')

    remoteSocket.close()

class DispatcherSkeleton(IDispatcher, ABC):
    #port 6969
    def run_skeleton(self):
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
        s.bind(("127.0.0.1", 6969))
        s.listen()

        while(True):
            (remoteSocket, addr) = s.accept()
            worker = multiprocessing.Process(target=Worker, args=(remoteSocket, self))
            worker.start()
        
